astropy/io/misc/yaml.py

```diff
@@ -0,0 +1,295 @@
+
+Example
+=======
+::
+
+  >>> from astropy.io.misc import yaml
+  >>> import astropy.units as u
+  >>> from astropy.time import Time
+  >>> from astropy.coordinates import EarthLocation
+
+  >>> t = Time(2457389.0, format='mjd',
+  ...          location=EarthLocation(1000, 2000, 3000, unit=u.km))
+  >>> td = yaml.dump(t)
+
+  >>> print(td)
+  !astropy.time.Time
+  format: mjd
+  in_subfmt: '*'
+  jd1: 4857389.5
+  jd2: 0.0
+  location: !astropy.coordinates.earth.EarthLocation
+    ellipsoid: WGS84
+    x: !astropy.units.Quantity
+      unit: &id001 !astropy.units.Unit {name: km}
+      value: 1000.0
+    y: !astropy.units.Quantity
+      unit: *id001
+      value: 2000.0
+    z: !astropy.units.Quantity
+      unit: *id001
+      value: 3000.0
+  out_subfmt: '*'
+  precision: 3
+  scale: utc
+
+  >>> ty = yaml.load(td)
+  >>> ty
+  <Time object: scale='utc' format='mjd' value=2457389.0>
+
+  >>> ty.location
+  <EarthLocation (1000.0, 2000.0, 3000.0) km>
+"""
+
+from __future__ import absolute_import
+
+import base64
+import numpy as np
+
+from ...time import Time, TimeDelta
+from ... import units as u
+from ... import coordinates as coords
+from ...utils.data_info import _get_obj_attrs_map
+
+try:
+    import yaml
+except ImportError:
+    raise ImportError('`import yaml` failed, PyYAML package is required for YAML')
+
+
+__all__ = ['AstropyLoader', 'AstropyDumper', 'load', 'load_all', 'dump']
+
+def _unit_representer(dumper, obj):
+    out = {'name': str(obj.to_string())}
+    return dumper.represent_mapping(u'!astropy.units.Unit', out)
+
+
+def _unit_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    return u.Unit(map['name'])
+
+
+def _time_representer(dumper, obj):
+    out = obj.info._represent_as_dict()
+    return dumper.represent_mapping(u'!astropy.time.Time', out)
+
+
+def _time_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    out = Time.info._construct_from_dict(map)
+    return out
+
+
+def _timedelta_representer(dumper, obj):
+    out = obj.info._represent_as_dict()
+    return dumper.represent_mapping(u'!astropy.time.TimeDelta', out)
+
+
+def _timedelta_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    out = TimeDelta.info._construct_from_dict(map)
+    return out
+
+
+def _ndarray_representer(dumper, obj):
+    if obj.flags['C_CONTIGUOUS']:
+        obj_data = obj.data
+    else:
+        cont_obj = np.ascontiguousarray(obj)
+        assert(cont_obj.flags['C_CONTIGUOUS'])
+        obj_data = cont_obj.data
+    data_b64 = base64.b64encode(bytes(obj_data))
+    out = dict(__ndarray__=data_b64,
+               dtype=str(obj.dtype),
+               shape=obj.shape)
+
+    return dumper.represent_mapping(u'!numpy.ndarray', out)
+
+
+def _ndarray_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    data = base64.b64decode(map['__ndarray__'])
+    return np.frombuffer(data, map['dtype']).reshape(map['shape'])
```

Comment:
I *think* (i.e., I didn't test...) that one should just be able to do
```
np.ndarray(buffer=data, shape=map['shape'], dtype=map['dtype'])
```
It might also be an idea to mimic what `np.save`/`np.load` do: the only difference would be to also just use the data if they are `F_CONTIGUOUS` and, if so, add an `order='F'` to the mapping.